crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -138,3 +155,84 @@ where
         .new_line("\n".to_string());
     ron::ser::to_string_pretty(&serialize, pretty_config)
 }
+
+#[cfg(test)]
+mod tests {
+    use bevy_app::AppTypeRegistry;
+    use bevy_ecs::{entity::EntityMap, system::Command, world::World};
+    use bevy_hierarchy::{AddChild, Parent};
+
+    use crate::dynamic_scene_builder::DynamicSceneBuilder;
+
+    #[test]
+    fn components_not_defined_in_scene_should_not_be_effected_by_scene_entity_map() {
+        // Testing that scene reloading applies EntitiyMap correctly to MapEntities components.
+
+        // First, we create a simple world with a parent and a child relationship
```

Comment:
Glad to see such a thorough test complete with helpful step-by-step comments that walk us through the intention behind the test. Thanks for this!